privacy-assessments/auto-expanding-details-privacy.md

```diff
@@ -0,0 +1,95 @@
+
+No.
+
+### 2.12. What temporary identifiers might this specification create or expose to the web?
+
+None.
+
+### 2.13. How does this specification distinguish between behavior in first-party and third-party contexts?
+
+If a match is found in a cross-origin iframe and a details element exists in a parent frame, this feature will not be used and the details element will not be automatically expanded.
+
+### 2.14. How does this specification work in the context of a user agent's Private Browsing or "incognito" mode?
+
+This feature will work the same regardless of incognito mode being enabled or disabled.
+
+### 2.15. Does this specification have a "Security Considerations" and "Privacy Considerations" section?
+
+I included a Privacy Considerations section further down this document. I did not include a Security Considerations section because I do not believe this feature exposes any security vulnerabilities.
+
+### 2.16. Does this specification allow downgrading default security characteristics?
+
+No.
+
+### 2.17. What should this questionnaire have asked?
+
+I have no suggestions to improve this questionnaire.
+
+## Privacy Considerations
+
+### 1. What privacy attacks have been considered?
+
+The page could observe the `toggle` event or the addition of the `open` attribute when this feature automatically expands details elements in an attempt to find out what the user was searching for with find-in-page or ScrollToTextFragment. If the page listens to the relevant events, it may be able to determine whether the user clicked on the details element to expand it or if it was automatically opened by this feature.
+
+### 2. What privacy attacks have been deemed out of scope (and why)?
+
+The aforementioned privacy attack is out of scope because it is less powerful than existing vulnerabilities for find-in-page and ScrollToTextFragment, meaning that this feature has a zero information delta with find-in-page and ScrollToTextFragment.
+
+For find-in-page, vulerabilities already exist which use the `scroll` event which is synchronous unlike the `toggle` event this feature will add. Every time find-in-page scrolls to a result, the page fires a `scroll` event. If the page creates a tiny scrollable area with every next possible character the user could type, it can recreate what the user typed into find-in-page. An example page which uses this attack can be found [here](/resources/find-in-page/search-incremental.html). If we were to remove the `scroll` events from find-in-page then this feature could become another vector for this attack, but this is unlikely to happen because it would break find-in-page for web pages which rely on the `scroll` event. This attack could be addressed regardless of the particular vector by adding a delay to find-in-page's scrolling which allows the user to type in enough characters to prevent the page incrementally building the user's search query, which Safari already does today.
```

Comment:
```suggestion
For find-in-page, vulnerabilities already exist which use the `scroll` event which is synchronous unlike the `toggle` event this feature will trigger. Every time find-in-page scrolls to a result, the page fires a `scroll` event. If the page creates a tiny scrollable area with every next possible character the user could type, it can recreate what the user typed into find-in-page. An example page which uses this attack can be found [here](/resources/find-in-page/search-incremental.html). If we were to remove the `scroll` events from find-in-page then this feature could become another vector for this attack, but this is unlikely to happen because it would break find-in-page for web pages which rely on the `scroll` event. This attack could be addressed regardless of the particular vector by adding a delay to find-in-page's scrolling which allows the user to type in enough characters to prevent the page incrementally building the user's search query, which Safari already does today.
```